src/libzfs/py_zfs_resource.c

```diff
@@ -235,6 +235,128 @@ PyObject *py_zfs_resource_get_properties(PyObject *self,
 	return py_zfs_get_properties(&res->obj, prop_set, get_source);
 }
 
+static
+PyObject *py_get_userprops(py_zfs_resource_t *res)
+{
+	PyObject *out = NULL;
+	nvlist_t *nvl = NULL;
+
+	Py_BEGIN_ALLOW_THREADS
+	PY_ZFS_LOCK(res->obj.pylibzfsp);
+
+	if (res->is_simple) {
+		zfs_refresh_properties(res->obj.zhp);
+		res->is_simple = B_FALSE;
+	}
+
+	nvl = zfs_get_user_props(res->obj.zhp);
+
+	PY_ZFS_UNLOCK(res->obj.pylibzfsp);
+	Py_END_ALLOW_THREADS
+
+	out = user_props_nvlist_to_py_dict(nvl);
+	return out;
+}
+
+PyDoc_STRVAR(py_zfs_resource_get_user_properties__doc__,
+"get_user_properties() -> dict\n"
```

Comment:
`get_user_properties(*)`

Reply:
The `*` character in a function signature has a special meaning. It's the separator between position / keyword arguments and arguments that are keyword-only. Since this method takes no arguments, the signature is `get_user_properties()`. 

For example
```
Help on built-in function getpid in module posix:

getpid()
    Return the current process id.
```